packages/@lwc/perf-benchmarks/src/__benchmarks__/engine-dom/js-framework-benchmark/append-rows-1k.benchmark.js

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { runJsFrameworkBenchmark, WARMUP_COUNT } from '../../../utils/runJsFrameworkBenchmark.js';
+
+// Based on https://github.com/krausest/js-framework-benchmark/blob/6c9f43f/webdriver-ts/src/benchmarksPuppeteer.ts
+// See `benchAppendToManyRows()`
+runJsFrameworkBenchmark(
+    `dom/js-framework-benchmark/append-rows/10k`,
+    { benchmark, before, run, after },
+    {
+        async warmup({ run, clear }) {
+            for (let i = 0; i < WARMUP_COUNT; i++) {
+                await run();
+                await clear();
+            }
+            await run();
+        },
+        async execute({ add }) {
+            await add();
+        },
+    }
+);
```

Comment:
All of these tests are adapted from https://github.com/krausest/js-framework-benchmark/blob/6c9f43f62604faa42bc7ddce634fde5a2e8b6eb7/webdriver-ts/src/benchmarksPuppeteer.ts